### DISCUSSION TITLE
How to make `include-after-body` escape the `#quarto-content` div?

### DISCUSSION THREAD

**Discussion by dense-set** [quarto-dev/quarto-cli#6308] [Q&A]
### Description

Hi again, Quarto wizards!

I noted quite a peculiar observation: Quarto docs [say that](https://quarto.org/docs/output-formats/html-basics.html#includes) the `include-after-body` will embed the contents of the mentioned file "before the `</body>` tag in HTML". However, this is not quite correct, for the included HTML is still wrapped inside the all-encompassing `<div id="quarto-content" ...>`. This is unexpected for me, for I want the immediate parent of the embedded HTML to be `<body>`. (The `#quarto-content` div also makes the HTML to be contained in a grid, which is unwanted by me.) 

How can I achieve this?

**Context:** This popped up while I was trying to create a custom footer (in a separate HTML file to be later `include-after-body`ed) as opposed to what is rendered by [`page-footer`](https://quarto.org/docs/websites/website-navigation.html#page-footer) option. The `#quarto-content` div makes my HTML included in a grid, thus making it impossible to have the full viewport width, which is intended for the footer.

Interestingly, the `page-footer` does circumvent the above problem by rendering its contents outside of the `#quarto-content` div, as desired.

**Comment by mcanouil** [comment]
Take a look at the partials, see <https://quarto.org/docs/journals/templates.html#html-partials>.

**Reply by dense-set** [reply]
I'm unable to garner any helpful info from that... 🫤

**Reply by mcanouil** [reply]
Open the template.html file, add for example: `$include-top$` within the body, then add in your YAML frontmatter, the information for the new template and whatever you want for `include-top`.
<img width="1624" alt="image" src="https://github.com/quarto-dev/quarto-cli/assets/8896044/c77fe832-dd33-4a13-bd42-addf29d0b217">

**Reply by dense-set** [reply]
That worked! But how do I include an HTML file instead? `include-top: file.html` didn't work. I also tried writing the full path, and the single-/double-quoted variations of these as well.

**Comment by cderv** [comment]
Can you share more about what you are doing and in which context ? Single document ? Website ? 

This is a really simple example and the after-body is after the `div#quarto-content`

````markdown
---
format: 
  html:
    include-after-body: 
      text: |
        <p>dummy</p>
---

Content
````

````html
<div id="quarto-content" class="page-columns page-rows-contents page-layout-article">

<!-- removed all in between for posting here on GH -->

</div> <!-- /content -->
<p>dummy</p>




</body></html>
````

**Reply by dense-set** [reply]
Yes, you are correct. But I am trying to do `include-after-body: footer.html` rather than:
```yml
include-after-body:
  text: |
    [contents of the file]
```

And it is here, (that is, while trying to wrap the stuff in a file first) that I report the peculiar behaviour for.

**Reply by cderv** [reply]
I get exactly the same by using a file 
````markdown
---
format: 
  html:
    include-after-body: 
      file: footer.html
---

Content

````

with `footer.html` being 
```html
<p>dummy</p>
```

Could you share a full reproducible example (a github repo maybe) that does show the behavior you described ? There may be some interaction with projects features, or included file content.

**Reply by dense-set** [reply]
Yes, here it is: https://github.com/dense-set/quarto-lab.

**Reply by cderv** [reply]
Thanks - so this is because of the website. What you want is to add is not a footer to the page, but a footer to the website. 

Basically, Quarto will render each document as a usual `format: html`, and `include-after-body` works as expected for the document. It will include the content after any body content provided in the .qmd document. This is a document options. *

However, in case of website, there will be a body envelope we add so that there is navbar, sidebar, footer, and any other website layout. Th e after-body of the document does come before the footer of the website. 

And this is what the `page-footer` is about, it is why it works as you noticed. Can you explain again why `pagefooter` does not work for you ?

**Reply by dense-set** [reply]
I was looking to have a border (an image) around the footer, but the `page-footer` option doesn't support that.

**Reply by cderv** [reply]
> I was looking to have a border (an image) around the footer

It seems to me this is CSS rules, isn't it ? 

If you do 
````
  page-footer: 
    center: |
      Veniam ut dolore cillum laborum ad consectetur nisi aliquip et laborum elit. Aute minim mollit laborum proident cupidatat nostrud ex fugiat eiusmod. Nostrud adipisicing reprehenderit quis nulla eu voluptate voluptate aute mollit minim ex deserunt non id. Elit ea excepteur qui sunt voluptate in. 
      Ullamco dolore fugiat ullamco laborum. Ut dolor voluptate aliqua velit voluptate ea est nostrud incididunt excepteur officia sit tempor. 
      Laborum culpa aute duis pariatur est occaecat cillum nisi ex velit excepteur.
````

and with your CSS you correctly get a border on your footer

![image](https://github.com/quarto-dev/quarto-cli/assets/6791940/487860ae-8ebc-4ea7-97e8-2e1c7f42c089)

Text is centered because right side is used for navigation item in some cases, and left side is also reserved. But you could use some CSS rules to override

````css
.nav-footer {
    display: block;
}
````

Though not using a flexbox could make some Quarto feature not working (like navigation menu for github source or something else). 

Hope it clarifies

**Reply by dense-set** [reply]
No, I want the top border to be an image, not a solid color.

**Reply by cderv** [reply]
Oh I see - you want a banner ? An image on top of the page-footer ?

**Reply by cderv** [reply]
I would say in that case you can probably add an image in the `page-footer`. This is HTML so you can use `<img>`. 

However you probably need to have some CSS tricks, and or JS tricks to make this works. 

It is not yet possible to tweak the default body enveloppe used for website using partials or a custom formats. 

@dragonstyle do you see any way to place an image on wide format in the footer above text content ?

**Reply by dense-set** [reply]
Yes, I want a banner-like image over the whole footer.

**Reply by mcanouil** [reply]
You can use CSS to add background image for the footer.
